providers/go-feature-flag/pkg/provider.go

```diff
@@ -0,0 +1,351 @@
+	case int:
+		v = int64(value)
+	default:
+		v = value
+	}
+
+	switch value := v.(type) {
+	case nil:
+		return model.GenericResolutionDetail[T]{
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				Reason:  of.Reason(rawResult.Reason),
+				Variant: rawResult.VariationType,
+			},
+		}
+	case T:
+		return model.GenericResolutionDetail[T]{
+			Value: value,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				Reason:  of.Reason(rawResult.Reason),
+				Variant: rawResult.VariationType,
+			},
+		}
+	default:
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewTypeMismatchResolutionError(fmt.Sprintf("unexpected type for flag %s", flagName)),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+}
+
+// evaluateWithRelayProxy is calling GO Feature Flag relay proxy to evaluate the file.
+func evaluateWithRelayProxy[T model.JsonType](provider *Provider, ctx context.Context, goffRequestBody model.EvalFlagRequest, flagName string, defaultValue T) model.GenericResolutionDetail[T] {
+	goffRequestBodyStr, err := json.Marshal(goffRequestBody)
+	if err != nil {
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError("impossible to marshal GO Feature Flag request"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	evalURL, err := url.Parse(provider.endpoint)
+	if err != nil {
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError("impossible to parse GO Feature Flag endpoint option"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	evalURL.Path = path.Join(evalURL.Path, "v1", "/")
+	evalURL.Path = path.Join(evalURL.Path, "feature", "/")
+	evalURL.Path = path.Join(evalURL.Path, flagName, "/")
+	evalURL.Path = path.Join(evalURL.Path, "eval", "/")
+
+	goffRequest, err :=
+		http.NewRequestWithContext(ctx, http.MethodPost, evalURL.String(), bytes.NewBuffer(goffRequestBodyStr))
+	if err != nil {
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError("error while building GO Feature Flag relay proxy request"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	goffRequest.Header.Set("Content-Type", "application/json")
+
+	response, err := provider.httpClient.Do(goffRequest)
+	if err != nil {
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError("impossible to contact GO Feature Flag relay proxy instance"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	responseStr, err := ioutil.ReadAll(response.Body)
```

Comment:
`ioutil.ReadAll` is deprecated, use `io.ReadAll` https://pkg.go.dev/io/ioutil#ReadAll